### ISSUE TITLE
State of Maintenance

### ISSUE THREAD

**Issue by sezanzeb** [sezanzeb/input-remapper#853]
You may have noticed that issues are piling up.

I believe most of them concern
- installation
- limitations of input-remapper/evdev
- figuring out how to do things

Thanks to the pretty high test coverage, I think the tool is rather stable luckily (Thinking back, I wish I applied the DI pattern everywhere, would have made writing tests easier.)

I have moved on to other hobbies though that are a bit less complicated quite some time ago (music and cycling among others), and I think @jonasBoss has also been doing different things than keeping an eye input-remapper for quite some time.

I am still looking at pull-requests, but don't expect valuable input from me in all other cases. Thanks to this being open-source, anyone is invited to participate in keeping this project alive.

Thanks for your support. I hope this post makes me feel better for turning my back on you.

**Comment by sezanzeb** [comment]
(I think we should probably remove the donation link from the front page since we aren't really involved anymore, @jonasBoss?)

**Comment by jonasBoss** [comment]
Well, given that I only read this now ...
I have been busy with university and working on my master thesis. I will try to be more active helping with issues but I don't have time to contribute much.

  
> I think we should probably remove the donation link from the front page since we aren't really involved anymore, @jonasBoss?

I agree

**Comment by ironhak** [comment]
Is there any active fork of this or valid alternatives?

**Comment by sezanzeb** [comment]
As you can see, there has been a release recently. I somehow ended up working on this quite a lot since autumn, after someone reported a problematic bug. And then I just kept seeing things that I wanted to change and add.

But I do hope this is only a temporary thing, because I don't even use input-remapper myself anymore.

Some people made donations, and those are greatly appreciated! I might have actually earned them. However, If I eventually stop making new commits once more, consider donating to someone else. Maybe to pipewire, mozilla, lineageos or KDE.

**Comment by sezanzeb** [comment]
Nope. Actually, all I wanted to do is to have a shift key on the extra buttons of my mouse, when I used to play a specific game. Haven't really played that one in, like, 3 years maybe.

**Comment by king-11** [comment]
Haha that makes sense every good software is born out of the developers own needs just like git :) I have to start thinking along those lines maybe try input remapper's rust re-write or something just for learning.

**Comment by nwildner** [comment]
@sezanzeb @jonasBoss , if you don't mind, I can work freely as a Support Engineer on this repo, the same way I did with [optimus-manager](https://github.com/Askannz/optimus-manager/issues/543#issuecomment-2199725558), tackling older issues that may not reproduce anymore or that are way far behind versions or that the initial reported lost interest.

LMK if you want this kind of help here. It can be overwhelming sometimes to have a queue that long and not being able to prioritize stuff. I'm actively using `input-remapper` with my [IINE Ungrade Keyboard+Mousepad](https://www.aliexpress.com/item/1005006588306103.html) so, I'm still interested on this.

Kind regards.

**Comment by sezanzeb** [comment]
Thanks for the offer! I used to look at every individual issue in detail, but eventually it has gotten "stable enough". At this point, honestly, I'm very rarely looking at issues anymore. Only to see if recent commits broke anything.

Sometimes I'm still in the loop and see a bug that can be reproduced, and then I may find time to fix it (For example the gamepad fixes draft PR). Or I get a new gamepad and realize input-remapper isn't handling something correctly myself.

So, I'm not sure if a clean/empty backlog would actually change much for me. I think it would just get bigger from that on again. Hm...

I'm also (unfortunately for everyone) not motivated enough to implement the possibly numerous feature requests that get posted every now and then.

I'd welcome people that contribute, but the architecture and code are very complex, due to my inexperience at the beginning, the quirks of the various kinds of input, security (Not running everything as root) and linux integration (udev, autostart, service, dbus)

I'm sure people would be happy to have someone look at their issues. But I'm afraid without an active code-conributing community it won't be very fulfilling, because I won't spend a lot of time on this.

Typically during winter, it seems, I do manage to allocate some time to work on things, for the most important problems: Emails from package maintainers, critical bugs, etc.

**Comment by sezanzeb** [comment]
There is https://github.com/users/sezanzeb/projects/5 by the way. I have kept track of a few things there.

Pull Requests that aren't continued by their original authors are regrettable. They are one of the more obvious things I'd work on in winter.

My todo list boils down to something like:
- critical or easily reproducible bugs
- open pull requests
- emails from package maintainers
- the rare case when implementing a new macro sounds cool

What I'm trying to say is, that I worry I'm not spending enough time on input-remapper anymore to honor the work you'd be putting into organizing the issue backlog.

**Comment by WesCook** [comment]
FYI, your project link seems to 404.  It may be marked as private on your end.

Nobody is expected to work on open-source forever.  I appreciate all the time+effort you've already put into this project.
